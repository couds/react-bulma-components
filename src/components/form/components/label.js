import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import modifiers from '../../../modifiers';

const Label = ({
  children,
  className,
  size,
  ...allProps
}) => {
  const props = modifiers.clean(allProps);
  return (
    <label
      {...props}
      className={classnames('label', modifiers.classnames(allProps), className, {
        [`is-${size}`]: size,
      })}
    >
      {children}
    </label>
  );
};

Label.propTypes = {
  ...modifiers.propTypes,
  children: PropTypes.node,
  className: PropTypes.string,
  style: PropTypes.shape({}),
  htmlFor: PropTypes.string,
  size: PropTypes.oneOf(['small', 'medium', 'large']),
};

Label.defaultProps = {
  ...modifiers.defaultProps,
  children: null,
  className: '',
  style: {},
  size: null,
  htmlFor: null,
};

export default Label;
