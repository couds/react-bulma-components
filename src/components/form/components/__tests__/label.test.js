import React from 'react';
import { render } from 'react-testing-library';
import { Label } from '../label';

describe('Label component', () => {
  it('should render', () => {
    const { asFragment } = render(<Label />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should concat classname in props with classname', () => {
    const { asFragment } = render(
      <Label className="other-class this-is-a-test">
        <p>Default</p>
      </Label>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should use inline styles', () => {
    const { asFragment } = render(
      <Label style={{ height: 250 }}>
        <p>Default</p>
      </Label>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
