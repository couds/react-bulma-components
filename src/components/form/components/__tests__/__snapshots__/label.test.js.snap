// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component should concat classname in props with classname 1`] = `
<DocumentFragment>
  <label
    class="label other-class this-is-a-test"
  >
    <p>
      Default
    </p>
  </label>
</DocumentFragment>
`;

exports[`Label component should render 1`] = `
<DocumentFragment>
  <label
    class="label"
  />
</DocumentFragment>
`;

exports[`Label component should use inline styles 1`] = `
<DocumentFragment>
  <label
    class="label"
    style="height: 250px;"
  >
    <p>
      Default
    </p>
  </label>
</DocumentFragment>
`;
